interceptors/logging/interceptors.go

```diff
@@ -50,64 +54,105 @@ func (c *reporter) PostCall(err error, duration time.Duration) {
+		requestLoggerDecision: requestLoggerDecision,
+		opts:                  r.opts,
+		logger:                r.logger.With(fields...),
+		kind:                  kind,
 	}, ctx
 }
 
 // UnaryClientInterceptor returns a new unary client interceptor that optionally logs the execution of external gRPC calls.
 // Logger will use all tags (from tags package) available in current context as fields.
-func UnaryClientInterceptor(logger Logger, opts ...Option) grpc.UnaryClientInterceptor {
+func UnaryClientInterceptor(logger Logger, decider PostRequestLoggingDecider, opts ...Option) grpc.UnaryClientInterceptor {
 	o := evaluateClientOpt(opts)
-	return interceptors.UnaryClientInterceptor(&reportable{logger: logger, opts: o})
+	return interceptors.UnaryClientInterceptor(&reportable{logger: logger, requestLoggerDecision: decider, opts: o})
 }
 
 // StreamClientInterceptor returns a new streaming client interceptor that optionally logs the execution of external gRPC calls.
 // Logger will use all tags (from tags package) available in current context as fields.
-func StreamClientInterceptor(logger Logger, opts ...Option) grpc.StreamClientInterceptor {
+func StreamClientInterceptor(logger Logger, decider PostRequestLoggingDecider, opts ...Option) grpc.StreamClientInterceptor {
 	o := evaluateClientOpt(opts)
-	return interceptors.StreamClientInterceptor(&reportable{logger: logger, opts: o})
+	return interceptors.StreamClientInterceptor(&reportable{logger: logger, requestLoggerDecision: decider, opts: o})
 }
 
 // UnaryServerInterceptor returns a new unary server interceptors that optionally logs endpoint handling.
 // Logger will use all tags (from tags package) available in current context as fields.
-func UnaryServerInterceptor(logger Logger, opts ...Option) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(logger Logger, decider PostRequestLoggingDecider, opts ...Option) grpc.UnaryServerInterceptor {
 	o := evaluateServerOpt(opts)
-	return interceptors.UnaryServerInterceptor(&reportable{logger: logger, opts: o})
+	return interceptors.UnaryServerInterceptor(&reportable{logger: logger, requestLoggerDecision: decider, opts: o})
 }
 
 // StreamServerInterceptor returns a new stream server interceptors that optionally logs endpoint handling.
 // Logger will use all tags (from tags package) available in current context as fields.
-func StreamServerInterceptor(logger Logger, opts ...Option) grpc.StreamServerInterceptor {
+func StreamServerInterceptor(logger Logger, decider PostRequestLoggingDecider, opts ...Option) grpc.StreamServerInterceptor {
 	o := evaluateServerOpt(opts)
-	return interceptors.StreamServerInterceptor(&reportable{logger: logger, opts: o})
+	return interceptors.StreamServerInterceptor(&reportable{logger: logger, requestLoggerDecision: decider, opts: o})
+}
+
+// TODO: yashrsharma44
+// What are the things that we want to log?
+func logRequestResponse(ctx context.Context, logger Logger, servObj interface{}, level Level) {
```

Comment:
Is it really response?

Reply:
From the client respect, the logging works the same manner, so it logs the response of the client.